degica/barcelona-cli
# Add vault auth

## Context

This PR adds logging in and using barcelona via vault authorization. With this change we will be able to login to barcelona instances that utilize vault for authorization.

## Known issue
No tests yet. Will add tests to the bits I have changed to the best of my ability

## How to test
1. As with most CLI changes, simply check that the CLI sends the right message. Run this command:

```
$ ./bcn -d login -a vault https://barcelona.localhost.labs.degica.com
```

2. Observe that something like this is returned:
```
Logging in with Vault
Create new GitHub access token with read:org permission here https://github.com/settings/tokens/new
GitHub Token:
POST /v1/login?debug=true HTTP/1.1
Host: barcelona.localhost.labs.degica.com
User-Agent: Go-http-client/1.1
Content-Length: 0
Accept: application/json
Content-Type: application/json
X-Vault-Token: somethingyoutyped
Accept-Encoding: gzip
```

Parts to note:
- It says `Logging in with Vault`
- The `X-Vault-Token` section is whatever you typed in the prompt for your Github Token.
